### PR TITLE
[Pspdfkit][MBL-14626] EZPZ Pspdfkit Update

### PR DESCRIPTION
**Pull request by Blackchai**
Only known  bug is the double creation of the "point"/stamp annotation type. I'm going to be resolving it in a separate ticket as it is not a regression.

### CODE REVIEW

**Review by JordanMarshall — CHANGES_REQUESTED** [review]

File: apps/teacher/src/main/java/com/instructure/teacher/PSPDFKit/AnnotationComments/AnnotationCommentListFragment.kt [hunk 1]
```diff
@@ -55,7 +55,7 @@ class AnnotationCommentListFragment : BaseListFragment<
                     annotation.contents = text
                     presenter.editComment(annotation, position)
                 }
-            }.show(fragmentManager, AnnotationCommentDialog::class.java.simpleName)
+            }.show(fragmentManager!!, AnnotationCommentDialog::class.java.simpleName)
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/teacher/src/main/java/com/instructure/teacher/PSPDFKit/AnnotationComments/AnnotationCommentListFragment.kt:58]: Oh that's right, :+1:

File: apps/teacher/src/main/java/com/instructure/teacher/activities/InitActivity.kt [hunk 2]
```diff
@@ -197,9 +198,9 @@ class InitActivity : BasePresenterActivity<InitActivityPresenter, InitActivityVi
         presenter.loadData(true)
     }
 
-    override fun onSaveInstanceState(outState: Bundle?) {
-        super.onSaveInstanceState(outState)
-        outState?.putInt(SELECTED_TAB, selectedTab)
+    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/teacher/src/main/java/com/instructure/teacher/activities/InitActivity.kt:201]: Oh perhaps that was the issue, kk, I'll try that.

File: apps/teacher/src/main/java/com/instructure/teacher/activities/SpeedGraderActivity.kt [hunk 3]
```diff
@@ -94,7 +95,7 @@ class SpeedGraderActivity : BasePresenterActivity<SpeedGraderPresenter, SpeedGra
 
     // This is here to prevent transaction too large exceptions, possibly caused by pdfFragment's saving.
     @SuppressLint("MissingSuperCall")
-    override fun onSaveInstanceState(outState: Bundle?) {}
+    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {}
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/teacher/src/main/java/com/instructure/teacher/activities/SpeedGraderActivity.kt:98]: :+1:

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/SettingsFragment.kt [hunk 4]
```diff
@@ -47,7 +47,7 @@ class SettingsFragment : BasePresenterFragment<ProfileSettingsFragmentPresenter,
         versionTextView.text = getString(R.string.fullVersion, BuildConfig.VERSION_NAME, BuildConfig.VERSION_CODE)
         profileButton.onClick { RouteMatcher.route(requireContext(), Route(ProfileFragment::class.java, null)) }
         rateButton.onClick { RatingDialog.showRateDialog(requireActivity(), com.instructure.pandautils.utils.AppType.TEACHER) }
-        legalButton.onClick { LegalDialog().show(fragmentManager, LegalDialog.TAG) }
+        legalButton.onClick { LegalDialog().show(fragmentManager!!, LegalDialog.TAG) }
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/teacher/src/main/java/com/instructure/teacher/fragments/SettingsFragment.kt:50]: :+1:

**Review by JordanMarshall — APPROVED** [review]

**Pull request merged**